orchagent/neighorch.cpp

```diff
@@ -354,13 +354,31 @@ void NeighOrch::doTask(Consumer &consumer)
             if (m_syncdNeighbors.find(neighbor_entry) == m_syncdNeighbors.end() || m_syncdNeighbors[neighbor_entry] != mac_address)
             {
                 if (addNeighbor(neighbor_entry, mac_address))
+                {
                     it = consumer.m_toSync.erase(it);
+                }
                 else
+                {
                     it++;
+                    continue;
+                }
             }
             else
+            {
                 /* Duplicate entry */
                 it = consumer.m_toSync.erase(it);
+            }
+
+            /* Remove remaining DEL operation in m_toSync for the same neighbor.
+             * Since DEL operation is supposed to be executed before SET for the same neighbor [ref: https://github.com/Azure/sonic-swss/pull/1184]
+             * A remaining DEL after the SET operation means the DEL operation failed previously and should not be executed anymore
```

Comment:
What if a  DEL is indeed coming after SET but not the previous failed one?

Reply:
According to the design in #1184, when a DEL operation comes later than SET, orch will remove the SET and only keep the DEL. Therefore, DEL+SET happens if and only if DEL is expected to be executed earlier. The scenario where there are both DEL and SET operations in m_toSync and the DEL operation should be executed later shall never happen. This is also why I referred to the PR in the comment.